cmd/kaf/consume.go

```diff
@@ -4,63 +4,113 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"math"
 	"os"
+	"sort"
 	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"text/tabwriter"
 	"time"
 
 	"github.com/Shopify/sarama"
 	"github.com/birdayz/kaf/avro"
 	"github.com/birdayz/kaf/pkg/proto"
 	"github.com/golang/protobuf/jsonpb"
-	prettyjson "github.com/hokaccha/go-prettyjson"
-	colorable "github.com/mattn/go-colorable"
+	"github.com/hokaccha/go-prettyjson"
+	"github.com/mattn/go-colorable"
 	"github.com/spf13/cobra"
 )
 
 var (
-	offsetFlag  string
-	raw         bool
-	follow      bool
-	schemaCache *avro.SchemaCache
-	keyfmt      *prettyjson.Formatter
+	partitionsWhitelistFlag []int
+	offsetFlags             []string
+	countFlags              []int
+	endFlag                 bool
+	raw                     bool
+	followFlag              bool
+	tailFlag                int
+	schemaCache             *avro.SchemaCache
+	keyfmt                  *prettyjson.Formatter
 
 	protoType string
 )
 
 func init() {
 	rootCmd.AddCommand(consumeCmd)
-	consumeCmd.Flags().StringVar(&offsetFlag, "offset", "oldest", "Offset to start consuming. Possible values: oldest, newest.")
-	consumeCmd.Flags().BoolVar(&raw, "raw", false, "Print raw output of messages, without key or prettified JSON")
-	consumeCmd.Flags().BoolVarP(&follow, "follow", "f", false, "Shorthand to start consuming with offset HEAD-1 on each partition. Overrides --offset flag")
-	consumeCmd.Flags().StringSliceVar(&protoFiles, "proto-include", []string{}, "Path to proto files")
-	consumeCmd.Flags().StringSliceVar(&protoExclude, "proto-exclude", []string{}, "Proto exclusions (path prefixes)")
-	consumeCmd.Flags().StringVar(&protoType, "proto-type", "", "Fully qualified name of the proto message type. Example: com.test.SampleMessage")
+	consumeCmd.Flags().StringSliceVarP(&offsetFlags, "offset", "o", []string{},
+		`Offset to start consuming. Pass a single value to apply to all partitions; pass a comma-delimited list to apply to specific partitions.
+Possible values: oldest, newest, -X, +X, X. (X is a number)
+oldest: Start reading at the oldest offset. This is the default.
+newest: Start reading at the newest offset.
+X: Start reading at offset X.
+-X: Start reading X before newest offset.
++X: Start reading X after oldest offset.`)
+	consumeCmd.Flags().IntSliceVarP(&countFlags, "count", "c", []int{}, "Number of messages to consume before exiting. Specify a single value for global limit, a comma-delimited list to limit specific partitions.")
```

Comment:
what do you think about rather having -n for this flag? i feel that this is more used within other tools, e.g. tail

Reply:
Makes sense to me.